cmd/login.go

```diff
@@ -0,0 +1,188 @@
+package cmd
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/user"
+	"strings"
+	"syscall"
+
+	"golang.org/x/crypto/ssh/terminal"
+
+	"github.com/spf13/cobra"
+	"github.com/turnerlabs/harbor-auth-client"
+)
+
+//Auth struc
+type Auth struct {
+	Version  string `json:"version"`
+	Username string `json:"username"`
+	Token    string `json:"token"`
+}
+
+// loginCmd represents the login command
+var loginCmd = &cobra.Command{
+	Use:   "login",
+	Short: "Login to harbor",
+	Long:  ``,
+	Run:   login,
+}
+
+var authURL = "https://auth.services.dmtio.net"
+
+func init() {
+	RootCmd.AddCommand(loginCmd)
+}
+
+func login(cmd *cobra.Command, args []string) {
+	_, _, err := Login()
+	if err != nil {
+		//log.Printf(err.Error())
+		fmt.Println("Login Failed")
+	}
+	return
+}
+
+func writeFile(version string, username string, token string) (bool, error) {
+	usr, err := user.Current()
+	if err != nil {
+		return false, err
+	}
+
+	var path = usr.HomeDir + "/.harbor"
+	err = os.Chdir(path)
+	if err != nil {
+		err = os.Mkdir(path, 0700)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	auth := new(Auth)
+	auth.Version = version
+	auth.Username = username
```

Comment:
Noticed that a `\n` is being appended to username in the credentials file.  Not sure if that matters or not...just looks odd.

```
{
  "version": "v1",
  "username": "jritsema\n",
  "token": "Y4xSSVi6jcmfMx9DaQqcpFbABskWx"
}
```